### PR TITLE
Upgrade to Swift 4.1

### PR DESCRIPTION
**Pull request by shagedorn**
For details, please see commit notes.

**Note:** Must not be merged before Xcode 9.3 is out of beta, as the changes require Swift 4.1.

### CODE REVIEW

**Comment by kishikawakatsumi** [comment]
@shagedorn Thank you for Pull Request. I commented some change request. Can you please fix them?

**Comment by shagedorn** [comment]
Are you sure you submitted your review? I cannot see any comments other than your summary.

**Comment by kishikawakatsumi** [comment]
@shagedorn I forgot to submit reviews, sorry.

File: Lib/KeychainAccess/Keychain.swift [hunk 1]
```diff
@@ -778,7 +778,7 @@ public final class Keychain {
     }
 
     public func allKeys() -> [String] {
-        return type(of: self).prettify(itemClass: itemClass, items: items()).flatMap { $0["key"] as? String }
+        return type(of: self).prettify(itemClass: itemClass, items: items()).compactMap { $0["key"] as? String }
```
[the remark this answers is not in the document]
Comment (shagedorn) [reply Lib/KeychainAccess/Keychain.swift]: I added a version switch that allows compiling with either Xcode 9.2 or 9.3, both without warnings. Is this an acceptable fix?

See: https://github.com/kishikawakatsumi/KeychainAccess/pull/371/commits/7d5c68e9824d8f268e3e31c8e9199fe33592c84b

File: Lib/KeychainAccess.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -391,13 +391,17 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 148E44E61BF9EDCB004FFEC1 /* Debug.xcconfig */;
 			buildSettings = {
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
```
[the remark this answers is not in the document]
Comment (shagedorn) [reply Lib/KeychainAccess.xcodeproj/project.pbxproj]: Thanks for the hint.

Reverted parts of these changes, as explained in the commit notes.

**Comment by shagedorn** [comment]
I don't really understand why my latest changes affect code coverage so severely. Any hints appreciated.

**Comment by shagedorn** [comment]
@kishikawakatsumi do you have any feedback on this one?

**Comment by kishikawakatsumi** [comment]
Xcode 9.3 is out. I will merge these changes soon, thank you! @shagedorn

**Comment by shagedorn** [comment]
…feel free to take a look why the code coverage has dropped so badly, I cannot really explain this.

**Comment by kishikawakatsumi** [comment]
@shagedorn This PR is superseded by https://github.com/kishikawakatsumi/KeychainAccess/pull/376. I added some commit to your PR. Thank you very much for contributing!!

**Pull request closed without merging**